src/components/MultiSelect/index.js

```diff
@@ -0,0 +1,240 @@
+
+const MultiSelect = props => {
+    const {
+        className,
+        style,
+        label,
+        hideLabel,
+        placeholder,
+        error,
+        bottomHelpText,
+        required,
+        disabled,
+        readOnly,
+        tabIndex,
+        variant,
+        chipVariant,
+        value,
+        onChange,
+        onFocus,
+        onBlur,
+        children,
+    } = props;
+    const [isOpen, setIsOpen] = useState(false);
+
+    const inputRef = useRef();
+    const triggerRef = useRef();
+    const dropdownRef = useRef();
+
+    const inputId = useUniqueIdentifier('input');
+    const errorMessageId = useErrorMessageId();
+
+    useEffect(() => {
+        if (isOpen) {
+            dropdownRef.current.focus();
+        }
+    }, [isOpen]);
+
+    const handleChange = val => {
+        inputRef.current.focus();
+        setIsOpen(false);
+        return onChange(val);
+    };
+
+    const handleDelete = option => {
+        if (Array.isArray(value)) {
+            return onChange(value.filter(val => val !== option));
+        }
+        return onChange([]);
+    };
+
+    const handleTriggerClick = () => {
+        if (disabled || readOnly) return;
+        setIsOpen(!isOpen);
+        if (!isOpen) {
+            // eslint-disable-next-line no-use-before-define
+            startListeningOutsideClick();
+        }
+    };
+
+    const handleKeyDown = event => {
+        if (event.target !== inputRef.current) return;
+        if (event.keyCode === ENTER_KEY || event.keyCode === SPACE_KEY) {
+            event.preventDefault();
+            if (disabled || readOnly) return;
+            setIsOpen(!isOpen);
+            if (!isOpen) {
+                // eslint-disable-next-line no-use-before-define
+                startListeningOutsideClick();
+            }
+        }
+    };
+
+    const handleOutsideClick = event => {
+        if (
+            event.target !== triggerRef.current.buttonRef.current &&
+            !triggerRef.current.buttonRef.current.contains(event.target)
+        ) {
+            // eslint-disable-next-line no-use-before-define
+            stopListeningOutsideClick();
+            setIsOpen(false);
+        }
+    };
+    const [startListeningOutsideClick, stopListeningOutsideClick] = useOutsideClick(
+        dropdownRef,
+        handleOutsideClick,
+    );
+
+    const chips = getChips(value, chipVariant, handleDelete);
+    const haveChips = chips && chips.length > 0;
+
+    return (
+        <StyledContainer className={className} style={style}>
+            <Label label={label} hideLabel={hideLabel} required={required} inputId={inputId} />
+            <StyledInput
+                variant={variant}
+                error={error}
+                disabled={disabled}
+                onFocus={onFocus}
+                onBlur={onBlur}
+                onKeyDown={handleKeyDown}
+                tabIndex={tabIndex}
+                ref={inputRef}
+            >
```

Comment:
we should follow accessibility from here:
https://www.lightningdesignsystem.com/components/combobox/#Multiple-selections
take a careful look to the markup used and all the aria, role and other attributes used